src/btree/bt_sync.c

```diff
@@ -100,6 +100,22 @@ __sync_dup_walk(WT_SESSION_IMPL *session, WT_REF *walk, uint32_t flags, WT_REF *
     return (0);
 }
 
+/*
+ * __wt_ref_is_obsolete --
+ *     Check whether the ref is obsolete according to the newest stop time pair.
+ */
+static void
+__wt_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref, bool *obsoletep)
+{
+    WT_ADDR *addr;
+
+    *obsoletep = false;
+
+    addr = ref->addr;
+    if (addr && __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts))
+        *obsoletep = true;
+}
+
```

Comment:
This function is static to the file and has no error return value, so I'd suggest:
```
/*
 * __sync_ref_is_obsolete --
 *     Return whether the ref is obsolete according to the newest stop time pair.
 */
static bool
__sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref) 
{
    WT_ADDR *addr;

    addr = ref->addr;

    return (addr != NULL && 
        __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts));
}
```

Regardless of what you choose, `addr` isn't a boolean, so needs an explicit comparison against `NULL`.

Reply:
I changed the function as per your suggestion. Later this function gets modified to include the pages that are in-memory and obsolete.  These will be handled in another ticket. Currently, the in-memory pages are ignored and they never are deleted until they get evicted. Whenever an in-memory page is found obsolete, planning to add them to urgent eviction queue and later when the checkpoint is happening on history store, those pages get removed while traversing the internal page of history store.